packages/astro/src/content/vite-plugin-content-imports.ts

```diff
@@ -297,3 +352,57 @@ async function getContentConfigFromGlobal() {
 
 	return contentConfig;
 }
+
+async function getDataEntryModule({
+	fileId,
+	dataEntryExtToParser,
+	contentPaths,
+	fs,
+	pluginContext,
+	settings,
+}: {
+	fs: typeof fsMod;
+	fileId: string;
+	contentPaths: Pick<ContentPaths, 'contentDir'>;
+	pluginContext: PluginContext;
+	dataEntryExtToParser: Map<string, DataEntryType['getEntryInfo']>;
+	settings: Pick<AstroSettings, 'config'>;
+}): Promise<DataEntryModule> {
+	const contentConfig = await getContentConfigFromGlobal();
+	const rawContents = await fs.promises.readFile(fileId, 'utf-8');
```

Comment:
I/O operations are always a hazard... we should always assume that they could throw an error :(

Reply:
Ugh, exceptions are so hard to track down in JS... I can add something here